### ISSUE TITLE
The generated `type Json = ColumnType<JsonValue, string, string>` should just be `JsonValue` for Postgres.

### ISSUE THREAD

**Issue by afg419** [RobinBlomberg/kysely-codegen#124]
If I'm not mistaken, the generated json type `type Json = ColumnType<JsonValue, string, string>` seems to be overly restrictive for Postgres, as I can use kysely to insert and update `JsonValue` values, in addition to strings. Since `JsonValue | string` is just `JsonValue`, the whole thing can be collapsed `type Json = JsonValue`.

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/RobinBlomberg) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/RobinBlomberg/kysely-codegen/issues/124">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/124/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/124/pledge.svg">
</picture>
</a>

**Comment by afg419** [comment]
That is, both of these work:
```
await kysely.insertInto('myTable').values({ address: { street: '111 First Ave', zip: '00000', state: 'MA' }}).executeTakeFirst()

await kysely.insertInto('myTable').values({ address: JSON.stringify({ street: '111 First Ave', zip: '00000', state: 'MA' })}).executeTakeFirst()
```

**Comment by BoscoDomingo** [comment]
I came here to post this exact issue. I changed the generated types and it works flawlessly (at least for insertion).

In fact, it works better than inserting a string because (at least in JS/TS) stringifying an object escapes the double quotes, therefore taking up significantly more space than needed. I wouldn't be surprised if that also causes us to lose the SQL optimizations and indexing abilities.

In essence, I echo the OP
